### DISCUSSION TITLE
ArgoCD instead of Flux

### DISCUSSION THREAD

**Discussion by cloudcompute** [zarf-dev/zarf#1897] [Q&A] [answered]
Hi Zarf team,

Greetings !

I looked at the Zarf's code, it has a BigBang extension that supports FluxCD. `Instead of Flux`, if I want to `use ArgoCD`, what do I need to do, is there any support for it?

Also, I had a look at the Terraform modules. Which modules shall I use to `provision an EKS cluster`.. I am listing them, kindly confirm. are these production-ready?

a. terraform-aws-uds-eks 
b. terraform-aws-uds-s3
c. terraform-aws-uds-s3-irsa
d. terraform-aws-tfstate-backend
e. terraform-aws-tfstate-vpc

Looks like there is `redundant code` between the `above-mentioned repositories` and `delivery-aws-iac`. Do I need to use delivery-aws-iac?

Thank you
Raman

**Comment by Racer159** [comment, marked as answer]
Apologies for the delay - most of our focus is in slack if you are able to join there! (https://zarf.dev/slack)

As of v0.29.0 Zarf supports ArgoCD in the same way as it does for Flux so it would function the same regardless of GitOps tooling (with the Agent mutating the git urls and secrets).

Currently we are still in the process of splitting out code from our `delivery` repos to the `uds` repos as we normalize IaC more for ourselves (hence the redundancy you noted) - currently `uds` isn't a fully released so while you could use those modules they aren't expressly production ready for public. There are other ways to spin up clusters as well without Terraform depending on your needs and you could also look at the example EKS package we have in the Zarf repo: https://github.com/defenseunicorns/zarf/tree/main/packages/distros/eks

For the time being though (until we standardize things) any of those will likely require some tweaking to fit your environment/use case.
